src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -80,18 +78,199 @@ internal class DefaultRazorSemanticTokensInfoService : RazorSemanticTokensInfoSe
 
             var semanticVersion = await GetDocumentSemanticVersionAsync(documentSnapshot).ConfigureAwait(false);
 
-            // If we have a matching cached response, avoid computation and return early.
-            if (_cachedResponses.TryGetValue(textDocumentIdentifier.Uri, out var response) &&
-                response.IsCSharpFinalized &&
-                response.SemanticVersion == semanticVersion &&
-                response.Range == range)
+            // The LSP client shouldn't send us ranges that start or end midway through a line, but we'll be
+            // defensive here since our tokens cache is only meant to be used with complete lines.
```

Comment:
if they shoudln't do this, can we error instead.  i really hate papering over bad requests as that just means they coudl cahnge their behavior, and users coudl see experience tank, and we'd have no idea.

Reply:
I'll rephrase the comment in my next push. It's not that they "shouldn't", but we wouldn't be able to utilize our cache if they send us partial line ranges. The added logging should be able to help us track down the problem if users experience issues